core/app/models/concerns/spree/user_methods.rb

```diff
@@ -72,8 +72,10 @@ def total_available_store_credit
     end
     deprecate total_available_store_credit: :available_store_credit_total, deprecator: Spree::Deprecation
 
-    def available_store_credit_total(currency:)
-      store_credits.reload.to_a.
+    def available_store_credit_total(currency:, force_reload: true)
+      # This reload makes eager loading not to work when hitting read only endpoints
+      store_credits.reload if force_reload
+      store_credits.to_a.
```

Comment:
while we are at it, I think it may be worth, in a dedicated commit, to slightly optimize this method... I don't see any reason to not use ActiveRecord for selecting the store credits with the right currency:

`store_credits.where(currency: currency).sum(&:amount_remaining)`

what do you think?

Reply:
Not sure about it, the thing is that we're trying to re-use as much as possible all data pre-loaded with eager loading, and if we do it, it'd execute extra queries to the db.